core/src/main/scala/org/apache/spark/deploy/worker/DriverRunner.scala

```diff
@@ -66,6 +66,22 @@ private[spark] class DriverRunner(
     def sleep(seconds: Int): Unit = (0 until seconds).takeWhile(f => {Thread.sleep(1000); !killed})
   }
 
+  // Grab all the SSL settings from the worker configuration
+  private val workerSecurityProps =
```

Comment:
Feels like the right thing would be to update `SparkConf.isExecutorStartupConf` and use it here.